addOns/client/src/main/java/org/zaproxy/addon/client/internal/ClientSideComponent.java

```diff
@@ -124,4 +122,57 @@ public boolean equals(Object obj) {
                 && Objects.equals(tagName, other.tagName)
                 && Objects.equals(text, other.text);
     }
+
+    @Override
+    public int compareTo(ClientSideComponent other) {
+        int result = 0;
+        result = stringCompare(this.getTypeForDisplay(), other.getTypeForDisplay());
```

Comment:
- Should assign directly.
 - I don't think we should be comparing with i18n'ed strings, this will change the order depending on the locale.

Reply:
It's what Simon asked for.

Reply:
⭐